src/Discord.Net.Rest/Entities/Messages/RestUserMessage.cs

```diff
@@ -148,6 +148,14 @@ public Task ModifySuppressionAsync(bool suppressEmbeds, RequestOptions options =
             TagHandling roleHandling = TagHandling.Name, TagHandling everyoneHandling = TagHandling.Ignore, TagHandling emojiHandling = TagHandling.Name)
             => MentionUtils.Resolve(this, 0, userHandling, channelHandling, roleHandling, everyoneHandling, emojiHandling);
 
+        public async Task PublishAsync(RequestOptions options = null)
+        {
+            if (Channel.GetType() == typeof(RestNewsChannel))
```

Comment:
Don't think failing silently is a behavior that's accepted.
There should be a check and throw an exception.
The check could also be changed to `Channel is RestNewsChannel`.

Reply:
Threw InvalidOperation - seemed the most logical.